feat(ciapp): add unittest ITR support (#20332)
DataDog/documentation: content/en/continuous_integration/intelligent_test_runner/setup/python.md

  * From `ddtrace>=2.1.0`.
  * From `Python>=3.7`.
  * Requires `coverage>=5.5`.
* `unittest`

>   * From `ddtrace>=2.2.0`
>   * From `Python>=3.7`

Reviewer: ```suggestion
  * From `ddtrace>=2.2.0`.
  * From `Python>=3.7`.
```

DD_ENV=ci DD_SERVICE=my-python-app DD_CIVISIBILITY_ITR_ENABLED=true pytest --ddtrace
{{< /code-block >}}

## Run unittest tests with the Intelligent Test Runner enabled

<div class="alert alert-info">Setting <code>DD_CIVISIBILITY_ITR_ENABLED</code> to true is required while the Intelligent Test Runner support for unittest is in beta. </div>

To run tests on services with the Intelligent Test Runner enabled, set `DD_CIVISIBILITY_ITR_ENABLED` to true.

`DD_CIVISIBILITY_ITR_ENABLED` (Optional)

> : Enable the Intelligent Test Runner coverage and test skipping features<br />

Reviewer: ```suggestion
: Enable the Intelligent Test Runner coverage and test skipping features.<br />
```

* `pytest`

>   * From `ddtrace>=1.19.0`

Reviewer: ```suggestion
  * From `ddtrace>=1.19.0`.
```

* `unittest`

>   * From `ddtrace>=TBD`

Reviewer: Reminder for this to be updated!

```suggestion
  * From `ddtrace>=TBD`.
```

**Note**: This does not override any other `skip` marks, or `skipif` marks that have a `condition` evaluating to `True`.

### Marking tests as unskippable in `unittest`

You can use [`unittest`][5]'s [`skipif` mark][6] to prevent the Intelligent Test Runner from skipping individual tests. Specify the `condition` as `False`, and the `reason` as `"datadog_itr_unskippable"`.

#### Individual tests

Individual tests can be marked as unskippable using the `@unittest.skipif` decorator as follows:
```python
import unittest

class MyTestCase(unittest.TestCase):
  @unittest.skipIf(False, reason="datadog_itr_unskippable")
  def test_function(self):
      assert True
```

> **Note**: This does not override any other `skip` marks, or `skipIf` marks that have a `condition` evaluating to `True`.

Reviewer: ```suggestion

Using `@unittest.skipif` does not override any other `skip` marks, or `skipIf` marks that have a `condition` evaluating to `True`.
```
